string.go

```diff
@@ -112,13 +124,18 @@ func (self *StringParser) Parse(
 		result = []byte(string(utf16.Decode(u16s)))
 	}
 
-	// If a terminator is specified read up to that.
 	if self.options.Term != "" {
 		idx := bytes.Index(result, []byte(self.options.Term))
 		if idx >= 0 {
 			result = result[:idx]
 		}
 	}
 
+	// If a terminator is specified read up to that.
+	if self.options.TermExpression != nil {
+		// Evaluate the offset expression with the current scope.
+			return EvalLambdaAsString(self.options.TermExpression, scope)
```

Comment:
EvalLambdaAsString will evaluate the expression but now we need to actually take that and search as on line 128.

I suggest we do this above line 128 and just assign options.Term = EvalLambdaAsString() then let it search for it as normal.

It would also be nice to have a test for this.

Reply:
just added a test for term_exp. May need to refactor if we want to use  options.Term = EvalLambdaAsString()  direct.